lightly/api/api_workflow_upload_embeddings.py

```diff
@@ -41,7 +41,10 @@ def set_embedding_id_to_latest(self):
             self._embeddings_api.get_embeddings_by_dataset_id(
                 dataset_id=self.dataset_id
             )
-        self.embedding_id = embeddings_on_server[-1].id
+        if len(embeddings_on_server) == 0:
+            raise RuntimeError(f"There are no known embeddings for dataset_id {self.dataset_id}.")
+        # return first entry as the API returns newest first
+        self.embedding_id = embeddings_on_server[0].id
```

Comment:
The entries we get from the API are sorted by what? (Not timestamp and latest entry is last?)

Reply:
`.const embeddings = await db.collection<IMongoEmbeddingSummary>('embeddingSummaries').find({ datasetId: toMongoId(datasetId) }).sort({ createdAt: -1 }).toArray();` thats how its called in the DB.

Don't know how to test it properly. Maybe you could upload multiple embeddings and verify that it always returns the first created embedding? Then we know this works as expected

Reply:
I would unittest this in the typescript tests an ensure it there.

Reply:
Edit: I would unittest the order in typescript.

Reply:
verified